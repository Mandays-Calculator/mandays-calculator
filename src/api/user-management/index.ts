export { getUserList } from "./UserManagement";
export type {
  UserListResponse,
  AddUserParams,
  AddUserResponse,
  DeleteUserParam,
  DeleteUserResponse,
  GetUsersParam,
} from "./types";
