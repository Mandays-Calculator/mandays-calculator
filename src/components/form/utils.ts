import type { FormErrors } from "./types";
import { defaultTo, get } from "lodash";

export const getOption = <Type extends object>(
  path: keyof Type | "value" | "label",
): ((option: Type) => string) => {
  return (option: Type) => {
    return defaultTo(get(option, path), "");
  };
};

export const getFieldError = (
  errors: FormErrors | undefined,
  name: string,
): string | undefined => {
  if (errors) {
    return typeof get(errors, name) === "string" ? get(errors, name) : "";
  }
};
