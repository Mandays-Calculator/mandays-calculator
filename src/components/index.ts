export * from "./form/text-field";
export * from "./form/Form";
export * from "./title";
