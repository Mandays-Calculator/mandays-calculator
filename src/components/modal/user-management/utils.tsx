export const genders = [
  {
    label: "FEMALE",
    value: "FEMALE",
  },
  {
    label: "MALE",
    value: "MALE",
  },
  {
    label: "NON-BINARY",
    value: "NON_BINARY",
  },
  {
    label: "PREFER NOT TO SAY",
    value: "PREFER_NOT_TO_SAY",
  },
];

export const rolesData = [
  {
    label: "System Admin",
    value: "ROLE_SYS_ADMIN",
  },
  {
    label: "Sprint Manager",
    value: "ROLE_SPRINT_MANAGER",
  },
  {
    label: "User",
    value: "ROLE_USER",
  },
];
