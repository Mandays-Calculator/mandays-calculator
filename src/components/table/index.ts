export * from "./Table";
export * from "./styles";
export * from "./types";
