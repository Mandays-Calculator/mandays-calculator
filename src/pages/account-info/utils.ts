import { object, string } from "yup";

export const actInfoChangePassSchema = () =>
  object().shape({
    currentPassword: string().required(),
    newPassword: string().required(),
    confirmNewPassword: string().required(),
  });
