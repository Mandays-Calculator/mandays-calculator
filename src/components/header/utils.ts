import { UserProject } from "~/api/auth";

export const getProjectOptions = (projects: UserProject[]): SelectObject[] => {
  return projects
    .filter((project: UserProject) => project.active)
    .map((project) => ({
      label: project.name,
      value: project.projectId,
    }));
};
