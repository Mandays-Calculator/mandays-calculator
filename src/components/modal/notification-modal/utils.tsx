import { ReactElement } from "react";
import { ModalType } from "./types";

import Grid from "@mui/material/Grid";
import WarningIcon from "@mui/icons-material/Warning";
import Typography, { TypographyProps } from "@mui/material/Typography";

import { SvgIcon } from "~/components";

interface IdleTimerSectionProps {
  idleTimer: number;
  idleTimerSeconds: number;
}

const StyledTypography: React.FC<TypographyProps> = (
  props: TypographyProps
): ReactElement => (
  <Typography
    variant="inherit"
    fontSize="3.5rem"
    fontFamily="sans-serif"
    {...props}
  />
);

export const IdleTimerSection: React.FC<IdleTimerSectionProps> = ({
  idleTimer,
  idleTimerSeconds,
}: IdleTimerSectionProps): ReactElement => (
  <Grid container spacing={1}>
    <Grid item xs={1} marginX="7rem">
      <StyledTypography>{idleTimer}min</StyledTypography>
    </Grid>
    <Grid item>
      <StyledTypography>{idleTimerSeconds}secs</StyledTypography>
    </Grid>
  </Grid>
);

export const renderIcon = (type: ModalType): ReactElement => {
  switch (type) {
    case "error":
    case "warning":
    case "unauthorized":
      return <SvgIcon name="warning" $size={8} />;
    case "idleTimeOut":
      return <WarningIcon color="error" sx={{ fontSize: 40 }} />;
    case "success":
    default:
      return <WarningIcon color="success" />;
  }
};

export const TimerSection = ({
  idleTimer,
  idleTimerSeconds,
}: {
  idleTimer: number;
  idleTimerSeconds: number;
}): ReactElement => (
  <>
    <IdleTimerSection
      idleTimer={idleTimer}
      idleTimerSeconds={idleTimerSeconds}
    />
    <Grid container spacing={3}>
      <Grid item xs={12}>
        <Typography variant="h5" style={{ margin: "1rem" }}>
          Please click "Stay Logged On" to keep working; or click "Logout" to
          end your session now.
        </Typography>
      </Grid>
    </Grid>
  </>
);
