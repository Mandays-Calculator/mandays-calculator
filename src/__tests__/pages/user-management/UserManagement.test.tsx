import {
  cleanup,
  fireEvent,
  render,
  screen,
  waitFor,
} from "@testing-library/react";
import UserEvent from "@testing-library/user-event";
import ProviderWrapper from "~/__tests__/utils/ProviderWrapper";
import { Table } from "~/components/table";

import { UserManagement } from "~/pages/user-management/";
import { UserList } from "./utils/editUserValues";
import { userListColumns } from "~/pages/user-management/user-list/utils";
import { useTranslation } from "react-i18next";
// import { mockUseUserList } from "./utils/mockUserManagementApis";

jest.mock("react-i18next", () => ({
  ...jest.requireActual("react-i18next"),
  useTranslation: () => ({ t: (key: any) => key }),
}));

describe("View User Management", () => {
  const component = (
    <ProviderWrapper>
      <UserManagement />
    </ProviderWrapper>
  );
  afterEach(cleanup);
  test("Render User Management", async () => {
    const user = UserEvent.setup();
    render(component);
    await waitFor(() => {
      const title = screen.getByText(/User Management/i);
      expect(title).toBeInTheDocument();

      const addButton = screen.getByTestId("test-add-user-btn");
      user.click(addButton);
      expect(addButton).toBeInTheDocument();
    });
  });

  // test("Render User Management View List", async () => {
  //   const user = UserEvent.setup();

  //   expect(mockUseUserList).toHaveBeenCalled();

  //   render(
  //     <ProviderWrapper>
  //       <UserManagement />
  //     </ProviderWrapper>
  //   );

  //   await waitFor(() => {
  //     // expect(mockQueryUserListApi as jest.Mock).toHaveBeenCalledTimes(1);
  //     const editButton = screen.getByTestId("test-edit-user-btn");
  //     user.click(editButton);
  //     expect(editButton).toBeInTheDocument();
  //   });
  // });
  jest.mock("formik", () => ({
    useFormikContext: () => {
      return {
        values: UserList,
        setValues: jest.fn().mockImplementation(() => {
          return {
            UpdateFirstName: "",
            UpdateLastName: "",
            UpdateMiddleName: "",
            UpdateSuffix: "",
            UpdateGender: "",
            UpdateEmail: "",
            UpdateEmployeeId: "",
            UpdateOdcId: "",
            UpdateCareerStep: "",
            UpdateJoiningDate: "",
            UpdateProjectId: "",
            UpdateTeamId: "",
            UpdateRoles: [],
          };
        }),
      };
    },
  }));

  test("user management table and edit button test", async () => {
    const { t } = useTranslation();
    const { getByTestId, container } = render(
      <Table
        name={"user-list"}
        data={UserList}
        columns={userListColumns({
          t,
          onDeleteUser: jest.fn(),
          onEditUser: jest.fn(),
        })}
      />
    );
    expect(container.textContent).toMatch("ditalomartin@gmail.com");
    await waitFor(() => {
      const editBtn = getByTestId("test-edit-user-btn");
      // editBtn.onclick = () => mockOnEditUser();

      expect(editBtn).toBeInTheDocument();
      fireEvent.click(editBtn);
    });

    expect(container).toMatchSnapshot("user-management-table-edit");
  }, 12000);
});
