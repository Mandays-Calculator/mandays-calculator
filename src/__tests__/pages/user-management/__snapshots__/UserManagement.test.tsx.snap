// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View User Management user management table and edit button test: user-management-table-edit 1`] = `
<div>
  <div
    class="MuiStack-root css-1ucnu3w-MuiStack-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        aria-label="user-list"
        class="MuiTable-root css-rqglhn-MuiTable-root"
        role="table"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-l7ewl1-MuiTableRow-root"
            role="row"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-145yh06-MuiTableCell-root"
              colspan="1"
              role="columnheader"
              scope="col"
            >
              <div
                class="MuiBox-root css-18biwo"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  Name
                </span>
              </div>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-145yh06-MuiTableCell-root"
              colspan="1"
              role="columnheader"
              scope="col"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="MuiBox-root css-18biwo"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  Email
                </span>
              </div>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-145yh06-MuiTableCell-root"
              colspan="1"
              role="columnheader"
              scope="col"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="MuiBox-root css-18biwo"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  Joining Date
                </span>
              </div>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-145yh06-MuiTableCell-root"
              colspan="1"
              role="columnheader"
              scope="col"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="MuiBox-root css-18biwo"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  Roles
                </span>
              </div>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-145yh06-MuiTableCell-root"
              colspan="1"
              role="columnheader"
              scope="col"
            >
              <div
                class="MuiBox-root css-18biwo"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  Action
                </span>
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          role="rowgroup"
        >
          <tr
            class="MuiTableRow-root css-boip6d-MuiTableRow-root"
            role="row"
            style="cursor: default;"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-cd7196-MuiTableCell-root"
              role="cell"
            >
              Martin Berula Ditalo
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-cd7196-MuiTableCell-root"
              role="cell"
            >
              ditalomartin@gmail.com
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-cd7196-MuiTableCell-root"
              role="cell"
            >
              2023-11-14
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-cd7196-MuiTableCell-root"
              role="cell"
            >
              ROLE_SYS_ADMIN
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-cd7196-MuiTableCell-root"
              role="cell"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                data-testid="test-edit-user-btn"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-254bcz-MuiSvgIcon-root"
                  data-testid="svg-component"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <use
                    href="#icon-edit"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                data-testid="test-delete-user-btn"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-sba5k-MuiSvgIcon-root"
                  data-testid="svg-component"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <use
                    href="#icon-delete"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
